ipatests/test_integration/test_trust.py

```diff
@@ -993,3 +1032,166 @@ def test_server_option_with_unreachable_ad(self):
             self.master.run_command(['rm', '-f', ad_zone_file])
             tasks.configure_dns_for_trust(self.master, self.ad)
             self.remove_trust(self.ad)
+
+
+class TestNonPosixAutoPrivateGroup(BaseTestTrust):
+    """
+    Tests for auto-private-groups option with non posix AD trust
+    Related : https://pagure.io/freeipa/issue/8807
+    """
+    topology = 'line'
+    num_ad_domains = 1
+    num_clients = 1
+
+    def test_add_nonposix_trust(self):
+        tasks.configure_dns_for_trust(self.master, self.ad)
+        tasks.establish_trust_with_ad(
+            self.master, self.ad_domain,
+            extra_args=['--range-type', 'ipa-ad-trust'])
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_auto_private_groups_default_trusted_range(self, type):
+        """
+        Modify existing range for default trusted AD domain range
+        with auto-private-groups set as true/hybrid/false and test
+        user with no posix attributes.
+        """
+        self.mod_idrange_auto_private_group(type)
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        (uid, gid) = self.get_user_id(self.clients[0], nonposixuser)
+        if type == "true":
+            assert uid == gid
+        else:
+            test_group = self.client[0].run_command(["id", nonposixuser])
+            gid_str = "gid={0}(domain users@{1})".format(gid, self.ad.domain)
+            grp_str = "groups={0}(domain users@{1})".format(gid,
+                                                            self.ad.domain)
+            assert gid_str in test_group.stdout_text
+            assert grp_str in test_group.stdout_text
+            assert uid != gid
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_idoverride_with_auto_private_group(self, type):
+        """
+        Override ad trusted user in default trust view
+        and set auto-private-groups=[hybrid,true,false]
+        and ensure that overridden values takes effect.
+        """
+        test_uid = "99999999"
+        test_gid = "98888888"
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+
+        with self.set_idoverrideuser(nonposixuser):
+            self.mod_idrange_auto_private_group(type)
+            (uid, gid) = self.get_user_id(self.master, nonposixuser)
+            assert(uid == test_uid and gid == test_gid)
+        result = self.clients[0].run_command(
+            ['id', nonposixuser], raiseonerr=False)
+        with xfail_context(
+                (result.returncode > 0),
+                reason='https://bugzilla.redhat.com/show_bug.cgi?id=2010512'):
+            tasks.assert_error(result, "no such user")
```

Comment:
The use of xfail_context is different. You need to start by writing your test with the intended behavior + checks (as if there was no BZ):
```
result = self.clients[0].run_command(
            ['id', nonposixuser], raiseonerr=False)
assert result == 0
# Add other checks here
```
Ideally, the `id` command should succeed on the client and provide uid and gid as expected.

In a second step, take into account the BZ that prevents the expected behavior:
```
with xfail_context(<condition>, <reason>):
    result = self.clients[0].run_command(
                ['id', nonposixuser], raiseonerr=False)
    assert result == 0
```